### PR TITLE
Joe Kane JS quiz

### PR DESCRIPTION
**Pull request by josephkane**

### CODE REVIEW

File: quiz.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	for (var i = 0; i < x.height; i++) {																							// use "i" to increment/decrement the key values
+		tree += " ".repeat(x.height - i - 1) + x.character.repeat((2 * i) + 1) + '\n';	// based on the index of the for loop
+	}
+
+		console.log(tree);
+
+		console.log("object:", treeObject);
+}
+
+
```
Comment (NewEvolution) [inline quiz.js]: Convention is to have one blank new line at the end of files, all other blank lines at the end should be removed.

File: quiz.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	if (x.height === "" || x.character === "") {																		// if the value of the key "height" is empty OR
+		alert("Oops! I'm missing information, Please fill out the WHOLE form.");			// if the value of the key "character" is empty
+	}																																								// when this function is called, display this alert
+
+	for (var i = 0; i < x.height; i++) {																							// use "i" to increment/decrement the key values
+		tree += " ".repeat(x.height - i - 1) + x.character.repeat((2 * i) + 1) + '\n';	// based on the index of the for loop
+	}
+
+		console.log(tree);
```
Comment (NewEvolution) [inline quiz.js]: Lines 96 and 98 should have the same indentation as the if and for statements above them

File: quiz.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	console.log("character", treeObject.character);
+}
+
+
+function buildTree (x) {
+
+	var tree = "";															// create a variable "tree" that contains an empty string
+
+
+	if (x.height === "" || x.character === "") {																		// if the value of the key "height" is empty OR
```
Comment (NewEvolution) [inline quiz.js:32]: Even though this pops up an alert if both fields aren't filled, the code below still runs and it prints out an empty tree of the given height if the height box is filled.

File: quiz.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	}
+}
+
+var treeObject = {};															// create an empty object called "treeObject"
+
+function middle () {
+	treeObject.height = heightBox.value;						// create a key in treeObject called "height" and add to it the text in the height input box
+	treeObject.character = charBox.value;						// create a key in treeObject called "character" and add to it the text in the character input box
+	buildTree(treeObject)														// run the function "buildTree" with the treeObject object
+	console.log("height:", treeObject.height);
```
Comment (NewEvolution) [inline quiz.js]: As per the instructions, the output should be just the tree itself.  These informative logs and the object log at line 98 should be removed for final submission (though logging them during production for troubleshooting is an _awesome_ idea)

File: quiz.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+// var char = document.getElementById("character").value;
+// console.log("character:", char);
+
+// var tree = "";
+// console.log("tree:", tree);
+
+
+
+var heightBox = document.getElementById("treeHeight");		// target input box in HTML, make into a variable
```
Comment (NewEvolution) [inline quiz.js:2]: Something REALLY WEIRD is going on with your comments from here on....they're migrating really really far to the right of the page somehow

File: quiz.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+//       *                spaces= 6, symbol= 1  (0) + 1
+//      ***               spaces= 5, symbol= 3  (1) + 2
+//     *****              spaces= 4, symbol= 5  (2) + 3
+//    *******             spaces= 3, symbol= 7  (3) + 4
+//   *********            spaces= 2, symbol= 9  (4) + 5
+//  ***********           spaces= 1, symbol= 11 (5) + 6
+// *************          spaces= 0, symbol= 13 (6) + 7
+
+
+
```
Comment (NewEvolution) [inline quiz.js]: The following lines, however, should be removed.  You shouldn't have commented out code in anything you're looking to finalize/submit.  While it's a WIP, sure, but remove unused code before final submission.  If you want to hang on to them for reference, you can always make a commit immediately before removing them, so you can go back and check out that commit later to check them, before making the final commit for the pull request.

File: quiz.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+
+
```
Comment (NewEvolution) [inline quiz.js]: Generally huge hunks of comments like this is better suited to be in a README than inline in your code.  Not a huge issue that needs fixing for these quizzes/exercises, just a note for future work once you start doing your own thing that needs explaining.

File: index.html [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+			<p>How high? <input id="treeHeight" type="text" autofocus></p>
+			<p>Which character? <input id="character" type="text"></p>
+
+			<p><button id="submitButton" type="button">Grow your tree!</button></p>
+
+
+		<script src="quiz.js"></script>
+	</body>
+</html>
```
Comment (NewEvolution) [inline index.html]: No newline at end of file.

**Comment by NewEvolution** [comment]
Few minor issues and some formatting.  Very concise clean code for the tree creation itself.

File: quiz.js [hunk 9]
```diff
@@ -75,30 +24,21 @@ function middle () {
...
 
 function buildTree (x) {
 
-	var tree = "";															// create a variable "tree" that contains an empty string
-
-
 	if (x.height === "" || x.character === "") {																		// if the value of the key "height" is empty OR
 		alert("Oops! I'm missing information, Please fill out the WHOLE form.");			// if the value of the key "character" is empty
-	}																																								// when this function is called, display this alert
+	}	else {																																							// when this function is called, display this alert
```
Comment (NewEvolution) [inline quiz.js:34]: Should only be a single space between the `if` closing curly brace & the `else`

Comment (josephkane) [inline quiz.js:34]: @NewEvolution I'm not sure why this is, but in my text editor there is just a single space like there should be. But I see that there isn't on here. Why is that?

Comment (NewEvolution) [inline quiz.js:34]: Oh, in that case it's just  GitHub being weird about whitespace.  If it looks right on your end, it's right.  If that's the only change left you're good to merge.

**Pull request merged**